Make builder

rvkubiak opened the pull request:
Allows execution of simple Make targets such as targets that execute shell commands to prepare a workspace.  This can be useful when replacing existing Make builds with Google Container Builder configs.

Philmod commented:
Could you add an `example` directory ?

ImJasonH commented on make/cloudbuild.yaml:
```diff
@@ -0,0 +1,11 @@
+
+steps:
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['build', '-t', 'gcr.io/$PROJECT_ID/make', '-t', 'gcr.io/$PROJECT_ID/make', '.']
```
This line tags the image twice with the same tag, is that intentional?

rvkubiak replied:
Not intentional, fixed.

rvkubiak commented:
Updated to squash commits.
Change entrypoint from bash to make to be consistent with other builders.
Added examples directory.
Removed duplicate tagging in Docker build.
